viewers/nerf_viewer/bridge_server/bridge_server_subprocess.py

```diff
@@ -0,0 +1,85 @@
+import atexit
+import os
+import signal
+import socket
+import subprocess
+import sys
+import threading
+import time
+
+import server
+
+from typing import Optional
+
+
+def is_port_open(port: int):
+    # check whether the given port is open
+    try:
+        sock = socket.socket()
+        sock.bind(("", port))
+        sock.close()
+
+        return True
+    except OSError:
+        return False
+
+
+def get_free_port(default_port: int = None):
+    if default_port:
+        if is_port_open(default_port):
+            return default_port
+    sock = socket.socket()
+    sock.bind(("", 0))
+    port = sock.getsockname()[1]
+
+    return port
+
+
+def run_bridge_server_as_subprocess(
+    websocket_port: int,
+    zmq_port: Optional[int] = None,
+    ip_address: str = "127.0.0.1",
+):
+    # run bridge server as a sub-process
+    args = [sys.executable, "-u", "-m", server.__name__]
+
+    # find an available port for zmq
+    if zmq_port is None:
+        zmq_port = get_free_port()
+        print(f"Using ZMQ port: {zmq_port}")
+
+    args.append("--zmq-port")
+    args.append(str(zmq_port))
+    args.append("--websocket-port")
+    args.append(str(websocket_port))
+    args.append("--ip-address")
+    args.append(str(ip_address))
+
+    process = subprocess.Popen(args, start_new_session=True)
+
+    def cleanup(process):
+        process.kill()
+        process.wait()
+
+    def poll_process():
+        """
+        Continually check to see if the viewer bridge server process is still running and has not failed.
+        If it fails, alert the user and exit the entire program.
+        """
+        while process.poll() is None:
+            time.sleep(0.5)
+
+        message = "The bridge server subprocess failed."
+        cleanup(process)
```

Comment:
`process` in `cleanup` is passed by arg list, but in `poll_process` it looks like a global var. 
Considering that the `process` variable has a clear lifetime and these functions are strongly related to it, would it be better to rewrite this code as a class and access it through `self.sub_process`?